src/plugins/hidsim/hidsim.cpp

```diff
@@ -106,8 +106,60 @@
 
 #include <libdrakvuf/libdrakvuf.h>  /* eprint_current_time */
 #include "../private.h"             /* PRINT_DEBUG */
+#include "hid_injection.h"          /* hid_inject */
+#include "gui_monitor.h"            /* gui_reconstruct */
+
 #include "hidsim.h"
 
+/*
+ * Checks, if GUI reconstruction is supported on the system under
+ * investigation
+ */
+bool hidsim::check_platform_support(drakvuf_t drakvuf)
+{
+    win_build_info_t bi;
+    {
+        vmi_lock_guard vmi(drakvuf);
+        if (!vmi_get_windows_build_info(vmi.vmi, &bi))
+            return false;
+    }
+
+    if (bi.version == VMI_OS_WINDOWS_7)
+    {
+        PRINT_DEBUG("[HIDSIM] GUI reconstruction supported"
+            "on Windows 7\n");
+        return true;
+    }
+
+    PRINT_DEBUG("[HIDSIM] GUI reconstruction is NOT supported "
+        "on this guest system\n");
+    return false;
+}
+
+bool hidsim::prepare_gui_reconstruction(drakvuf_t drakvuf, const char* win32k_profile)
+{
+    if (!win32k_profile)
+    {
+        PRINT_DEBUG("[HIDSIM] No win32k-profile provided. Unable to monitor the GUI\n");
+        return false;
+    }
+
+    this->win32k_json_path = win32k_profile;
+
+    this->is_gui_support = this->check_platform_support(drakvuf);
+
+    page_mode_t pm = drakvuf_get_page_mode(drakvuf);
+    bool is_x86 = pm == VMI_PM_PAE ? true: false;
+
```

Comment:
bool is_x86 = pm == VMI_PM_PAE;
is enough :)

Reply:
Changed in f22d6d3460047e8db37afebe41201d46bfc24b90